bot.py

```diff
@@ -48,6 +49,23 @@ def __init__(self):
         self.register_command("dinkster", handler=self.dinkster)
         self.register_command("joke", handler=self.joke)
 
+        async def connect_deaf(self):
+            logging.info("Connecting to voice channel")
+            voice_client = await self.connect()
+            await voice_client.guild.change_voice_state(
+                channel=self,
+                self_deaf=True,
+            )
+            voice_states = voice_client.channel.voice_states
+            await asyncio.sleep(1)  # Without sleep deafening isn't registered in logs
+            logging.info(
+                "Bot is deafened: %s", voice_states[voice_client.user.id].self_deaf
+            )
+            return voice_client
+
+        discord.VoiceChannel.connect_deaf = connect_deaf
```

Comment:
Instead of this, how about just having connect_deaf as a method which accepts a VoiceChannel as an argument?